lib/CppInterOp/CppInterOp.cpp

```diff
@@ -4289,28 +4295,60 @@ class StreamCaptureInfo {
         }()},
         m_FD(FD) {
 #else
-  StreamCaptureInfo(int FD) : m_TempFile{tmpfile()}, m_FD(FD) {
+  StreamCaptureInfo(int FD) : mode(FD) {
+#endif
+#if !defined(CPPINTEROP_USE_CLING) && !defined(_WIN32)
+    auto& I = getInterp(NULLPTR);
+    if (I.isOutOfProcess() && FD == STDOUT_FILENO) {
+      m_TempFile = I.getTempFileForOOP(FD);
+      ::fflush(m_TempFile);
+      m_FD = fileno(m_TempFile);
+      m_OwnsFile = false;
+    } else {
+      m_TempFile = tmpfile();
+      m_FD = FD;
+      m_OwnsFile = true;
+    }
+#else
+    m_TempFile = tmpfile();
+    m_FD = FD;
+    m_OwnsFile = true;
+    (void)mode;
```

Comment:
I don't think we need to change anything for the current implementation of redirection.
You are currently redirecting `stdout` and `stderr` to the parent process's `stdout` and `stderr`. So, the current implementation of `StreamCaptureInfo` should, "in theory", just work out of the box. You will also be able to remove the `Interpreter::FileDeleter` and `Interpreter::IOContext` in that case.
`stdin` is a different matter, and I am not sure how that works. But this class (`StreamCaptureInfo`) does not deal with `stdin` anyway.

Let me know if I am wrong about not needing any changes here. Or you think this changes are required.